ethcore/types/src/receipt.rs

```diff
@@ -49,12 +49,12 @@ pub struct Receipt {
 
 impl Receipt {
 	/// Create a new receipt.
-	pub fn new(outcome: TransactionOutcome, gas_used: U256, logs: Vec<LogEntry>) -> Receipt {
-		Receipt {
-			gas_used: gas_used,
-			log_bloom: logs.iter().fold(Bloom::default(), |mut b, l| { b = &b | &l.bloom(); b }), //TODO: use |= operator
-			logs: logs,
-			outcome: outcome,
+	pub fn new(outcome: TransactionOutcome, gas_used: U256, logs: Vec<LogEntry>) -> Self {
+		Self {
+			gas_used,
+			log_bloom: logs.iter().fold(Bloom::default(), |b, l| b | l.bloom()),
```

Comment:
could you change `b | l.bloom()` to `b.accrue_bloom(l.bloom())` ? The second one does not allocate new bloom on stack, but modifies existing one